src/components/Messages/MessageText/markdown.js

```diff
@@ -29,6 +29,58 @@ md.renderer.rules.link_open = (tokens, idx, options, env, self) => {
 	return defaultRender(tokens, idx, options, env, self);
 };
 
+md.renderer.rules.paragraph_open = function(tokens, idx, options, env, self) {
+	let result = '';
+	if (idx > 1) {
+		let inline = tokens[idx - 2];
+		const paragraph = tokens[idx];
+		if (inline.map && inline.map[1] && paragraph.map && paragraph.map[0]) {
+			const diff = paragraph.map[0] - inline.map[1];
+			if (diff > 0) {
+				result = '<br>'.repeat(diff);
+			}
+		}
+
+		const listClose = tokens[idx - 1];
+		if (listClose.type.includes('list_close')) {
+			inline = tokens[idx - 4];
+			if (inline.map && inline.map[1] && paragraph.map && paragraph.map[0]) {
+				const diff = paragraph.map[0] - inline.map[1];
+				if (diff > 0) {
+					result = '<br>'.repeat(diff);
+				}
+			}
+		}
+	}
+	return result + defaultRender(tokens, idx, options, env, self);
+};
+
+md.renderer.rules.list_item_open = function(tokens, idx, options, env, self) {
```

Comment:
PR working as expected on my local. @AlexanderKanakis Is this implementation our own or from any reference? I'm sorry, It's new for me so just trying to understand what's going on. Why `list_item_open`? If the rules for both are almost the same(just the difference with some indices),  is creating common fn a good idea?

Reply:
@Shailesh351 `list_item_open` contains items in a list. A `list_item_open` item should contain a `paragraph_open` so we could edit only one ruleset, but checking both make it easier to handle newline insertion in-between list items and before the list.